File: CodeChef/long-challenge/OCT15/SUBINC.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
   	long long int t,n;
    scanf("%lld",&t);
    while(t--)
    {
    	scanf("%lld",&n);
    	long long int arr[n];
    	long long int e=1;
    	long long int count=0;
    	scanf("%lld",&arr[0]);
    	
    	for(int i=1;i<n;i++){
    		scanf("%lld",&arr[i]);
    		if(arr[i]>=arr[i-1]){
    				e++;
    				continue;
    		}
    		
    		count+=((e*(e+1))/2);
    		e=1;
    		

    	}
    	count+=((e*(e+1))/2);
    	printf("%lld\n",count );
    }

    return 0;
}

File: CodeChef/long-challenge/OCT15/SUBINC.py
t=int(input())
while t>0:
	t-=1
	n=int(input())
	arr=list(map(int,input().split()))
	count=0
	e=1
	for i in range(1,n):
		if(arr[i]>=arr[i-1]):
			e+=1
			continue
		count+=(e*(e+1))/2
		e=1
	count+=(e*(e+1))/2
	print(int(count))

